File: 品味Java的21種設計模式/dp_java/strategy/example4/CooperateCustomerStrategy.java
package cn.javass.dp.strategy.example4;
/**
 * 具体算法实现，为战略合作客户客户计算应报的价格
 */
public class CooperateCustomerStrategy implements Strategy{
	public double calcPrice(double goodsPrice) {
		System.out.println("对于战略合作客户，统一8折");
		return goodsPrice*0.8;
	}
}

File: 品味Java的21種設計模式/dp_cpp/strategy/example4/CooperateCustomerStrategy.cpp
#include "CooperateCustomerStrategy.h"

namespace cn {
namespace javass {
namespace dp {
namespace strategy {
namespace example4 {

double CooperateCustomerStrategy::calcPrice(double goodsPrice) {
    puts("对于战略合作客户，统一8折");
    return goodsPrice * 0.8;
}
}
}
}
}
}
